### ISSUE TITLE
Implement correct "resetOnFinished" on animation component

### ISSUE THREAD

**Issue by m3taphysics** [decentraland/unity-explorer#394]
### **Issue Description:**

The Animator component includes a property named shouldReset, this property in theory reverts the model to its initial pose once the animation finished playing. However, it doesn’t work, the property is ignored.

What makes this problem especially complicated is that the `playSingleAnimation()` helper sets this property to true by default, this is how most animations are triggered. So if we fix the shouldReset property, suddenly a lot of scenes will change behavior. This includes pretty disruptive things, like doors shutting right after they finish opening, or elevators snapping back to the ground after they’re done rising.

- [ ] Define a new `resetOnFinished` property, that does the same, so as to not break any existing scenes.
- [ ] The `shouldReset` property should not be implemented.

**Comment by nearnshaw** [comment]
I put together this minimal example scene to test out the functionality. In it I have two dogs, one should stay sitting, the other shouldn't. They use different values for the shouldReset property

[New Scene 3.zip](https://github.com/user-attachments/files/18512095/New.Scene.3.zip)

I was able to reproduce the issue on the new client, the dogs both sit and stay sitting, which is not the expected behavior:

https://github.com/user-attachments/assets/c35b9da3-9b48-44b7-b8cd-0cba7fbdddfb

HOWEVER, in the Web client I didn't get good results either.

If I used what should be the comfortable easy syntax, I saw the same wrong behavior as in the E@ 100% of the time
`Animator.playSingleAnimation(doggo2, "Sitting", true)``

https://github.com/user-attachments/assets/232554ec-14a8-4bdd-b594-c9bd8b97654b

Instead I had to use this more pure syntax, which is not the ideal
`Animator.getClip(doggo2, "Sitting").playing = true``

And doing that it only partially worked. It seems to reset only half way for some reason, and only about 50% of the time. The other times it behaves just like the new client

https://github.com/user-attachments/assets/10651d63-e3d1-4bd1-bbaa-bab9a7200f6d


In conclusion: 

- The auto-reset is a valuable feature, we want to have it to make creator's life easier
- BUT doesn't seem like it's a necessity to fix it for backwards compatibility, since it was already not working reliably. If a scene used it, it likely was having buggy behavior in the old client too. So unless there's a more reliable way to use it that I'm missing, I wouldn't add it to the priorities of this cycle.
- There seems to be a separate SDK-side issue. The fact that we don't have the same behavior with the two different syntax variations I show above suggests that something is not working at that level
